src/saml.ts

```diff
@@ -376,13 +387,27 @@ class SAML {
           "@xmlns:saml": "urn:oasis:names:tc:SAML:2.0:assertion",
           "#text": this.options.issuer,
         },
+        "samlp:Extensions": {},
```

Comment:
Why do you add the empty value here and then delete it later when not needed? The pattern you used previously just adds the value if you need it and skips the extra logic to delete if not needed.

Reply:
it is because strick type of NameID here https://github.com/node-saml/node-saml/blob/71d9fe5ace94e19f1f5f288f1093a331a9758471/src/types.ts#L60

and `samlp:Extensions` element should be placed before NameID. There is other way also but I choose and Like to keep this one for safe type.

Reply:
@cjbarth once you get time, please review it